test/e2e/serial/tests/configuration.go

```diff
@@ -453,25 +486,40 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 
 			defer func() {
 				By("reverting kubeletconfig changes")
-				Eventually(func(g Gomega) {
-					err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(targetedKC), targetedKC)
-					Expect(err).ToNot(HaveOccurred())
+				if len(targetedKC.OwnerReferences) == 0 {
+					Eventually(func(g Gomega) {
+						err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(targetedKC), targetedKC)
+						Expect(err).ToNot(HaveOccurred())
 
-					kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
-					Expect(err).ToNot(HaveOccurred())
+						kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+						Expect(err).ToNot(HaveOccurred())
 
-					kcObj.TopologyManagerScope = initialTopologyManagerScope
-					err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
-					Expect(err).ToNot(HaveOccurred())
+						kcObj.TopologyManagerScope = initialTopologyManagerScope
+						err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+						Expect(err).ToNot(HaveOccurred())
 
-					err = fxt.Client.Update(context.TODO(), targetedKC)
-					g.Expect(err).ToNot(HaveOccurred())
-				}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+						err = fxt.Client.Update(context.TODO(), targetedKC)
+						g.Expect(err).ToNot(HaveOccurred())
+					}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+				}
+				if len(targetedKC.OwnerReferences) == 1 {
+					initialAnnotations, err := json.Marshal(performanceProfile.Annotations)
+					Expect(err).ToNot(HaveOccurred())
+					err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(&performanceProfile), &performanceProfile)
+					Expect(err).ToNot(HaveOccurred())
 
+					By("Applying changes in performance profile and waiting until mcp will start updating")
+					klog.Infof("updating annotations from:\n%+v\nto:\n%+v\n", performanceProfile.Annotations, string(initialAnnotations))
+					Expect(fxt.Client.Patch(context.TODO(), &performanceProfile,
+						client.RawPatch(
+							types.JSONPatchType,
+							[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/metadata/annotations", "value": %s }]`, initialAnnotations)),
+						),
+					)).ToNot(HaveOccurred())
+				}
 				By("waiting for mcp to start updating")
 				waitForMcpsCondition(fxt.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdating)
 
-				By("wait for mcp to get updated")
 				waitForMcpsCondition(fxt.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdated)
```

Comment:
These lines are not inside the if statements and therefore if there is no change that happens to the initial state of the kubeletconfig or under the performance profile `waitForMcpCondition` will get stuck here and fail eventually.
So a better option would probably be to check if there even needs to be a change to the initial state if not it can all be skipped.

Reply:
right, I'll update this. Thanks for catching

Reply:
in fact getNewTopologyManagerScopeValue() ensures to return scope different from the initial one, nevertheless I added an assertion